### ISSUE TITLE
resizing window

### ISSUE THREAD

**Issue by dcnieho** [pthom/hello_imgui#28]
I have a window whose size i only know later, not on the first or the second frame. I need some advice how to deal with this with hello_imgui using Python. My window is real simple, it just has an image (`imgui.image()`) to show. I have the following questions:
1. How to size my window such that is has the size of the image, taking into account DPI scaling of the monitor? I think that `app_window_params.window_geometry.window_size_measure_mode = hello_imgui.WindowSizeMeasureMode.relative_to96_ppi` is key here, right?
2. How to only size the window once its size is know? The first image to show may arrive seconds after the window has been created. I tried hiding the window (`glfw.hide_window`) in `post_init` and only showing it once the image is about to be drawn for the first time. Can i invoke the sizing (and positioning) logic at a later time instead of only at window creation?`
3. Ideally i'd like the logic to scale the window to 100% image size if that fits on the monitor, else to clamp the window size down to the monitor size while maintaining aspect ratio. Is that possible?

**Comment by pthom** [comment]
Hi Dee,Those are interesting  subjects on which I worked a lot.I would be interested in seeing that on your computer, since it is a windows PC. Could we try to have a telephone conversation at 2 PM?Thanks Le 23 janv. 2023 à 10:46, Diederick C. Niehorster ***@***.***> a écrit :﻿
I have a window whose size i only know later, not on the first or the second frame. I need some advice how to deal with this with hello_imgui using Python. My window is real simple, it just has an image (imgui.image()) to show. I have the following questions:

How to size my window such that is has the size of the image, taking into account DPI scaling of the monitor? I think that app_window_params.window_geometry.window_size_measure_mode = hello_imgui.WindowSizeMeasureMode.relative_to96_ppi is key here, right?
How to only size the window once its size is know? The first image to show may arrive seconds after the window has been created. I tried hiding the window (glfw.hide_window) in post_init and only showing it once the image is about to be drawn for the first time. Can i invoke the sizing (and positioning) logic at a later time instead of only at window creation?`
Ideally i'd like the logic to scale the window to 100% image size if that fits on the monitor, else to clamp the window size down to the monitor size while maintaining aspect ratio. Is that possible?
